src/compiler/checker.ts

```diff
@@ -20964,6 +20985,25 @@ namespace ts {
                         flags |= ModifierFlags.Async;
                         lastAsync = modifier;
                         break;
+
+                case SyntaxKind.OverrideKeyword:

```

Comment:
you also need to add a check in `checkFlagAgreementBetweenOverloads` to make sure all overloads are marked as `override`.